libraries/AP_NavEKF2/AP_NavEKF2_Outputs.cpp

```diff
@@ -179,18 +179,20 @@ void NavEKF2_core::getWind(Vector3f &wind) const
 }
 
 
-// return NED velocity in m/s
+// return the NED velocity of the body frame origin in m/s
 //
 void NavEKF2_core::getVelNED(Vector3f &vel) const
 {
-    vel = outputDataNew.velocity;
+    // correct for the IMU position offset (EKF calculations are at the IMU)
+    vel = outputDataNew.velocity + velOffsetNED;
 }
 
-// Return the rate of change of vertical position in the down diection (dPosD/dt) in m/s
+// Return the rate of change of vertical position in the down diection (dPosD/dt) of the body frame origin in m/s
 float NavEKF2_core::getPosDownDerivative(void) const
 {
-    // return the value calculated from a complmentary filer applied to the EKF height and vertical acceleration
-    return posDownDerivative;
+    // return the value calculated from a complementary filter applied to the EKF height and vertical acceleration
+    // correct for the IMU offset (EKF calculations are at the IMU)
+    return posDownDerivative + velOffsetNED.z;
```

Comment:
Why do this instead of adding the position offset before differentiating? Won't this be noisier?